build-docker.sh

```diff
@@ -59,28 +76,37 @@ for BITCOIN_ONLY in 0 1; do
   DIRSUFFIX=${BITCOIN_ONLY/1/-bitcoinonly}
   DIRSUFFIX=${DIRSUFFIX/0/}
 
+  SCRIPT_NAME=".build_legacy_$BITCOIN_ONLY.sh"
+  cat <<EOF > "build/$SCRIPT_NAME"
+    # DO NOT MODIFY!
+    # this file was generated by ${BASH_SOURCE[0]} on $(date)
```

Comment:
We don't want to introduce any unnecessary non-determinism.

```suggestion
    # this file was generated by ${BASH_SOURCE[0]}
```